### PR TITLE
Adding basic flannel support

### PR DESCRIPTION
**Pull request by mihok**
Recreating the PR from #1608 as a new branch/PR. I totally blew up the original trying to rebase and then things got messy and out of control...

> Adding basic udp support for flannel overlay networking with kubernetes. Would like to pull the non-masqurading CIDR into the flannel yaml file in the future and be able to configure aws-vpc mode in flannel, but figured baby steps, get it in to kops first.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1863)

### CODE REVIEW

**Comment by justinsb** [comment]
@k8s-bot ok to test

File: upup/models/cloudup/resources/addons/networking.flannel/v0.7.0.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        tier: node
+        app: flannel
+        role.kubernetes.io/networking: "1"
+    spec:
+      hostNetwork: true
+      nodeSelector:
+        beta.kubernetes.io/arch: amd64
+      serviceAccountName: flannel
+      containers:
+      - name: kube-flannel
```
Comment (chrislovecnm) [inline upup/models/cloudup/resources/addons/networking.flannel/v0.7.0.yaml]: Can we get resource mem and cpu limits on a follow-up PR?

Comment (mihok) [inline upup/models/cloudup/resources/addons/networking.flannel/v0.7.0.yaml]: Absolutely, planning on two PRs: 
- Mem/CPU limits
- Pulling nonMasquradingCIDR from kops config

Comment (chrislovecnm) [inline upup/models/cloudup/resources/addons/networking.flannel/v0.7.0.yaml]: 👍

**Pull request merged**